src/System.Text.Encoding/tests/ASCIIEncoding/ASCIIEncodingDecode.netcoreapp.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Xunit;
+
+namespace System.Text.Tests
+{
+    public partial class ASCIIEncodingDecode
+    {
+        [Theory]
+        [InlineData("hello!", 6)]
+        [InlineData("hello\u1234there!", 16)]
+        [InlineData("\ud800\udc00", 10)]
+        public void GetByteCount_WithReplacementFallback(string input, int expectedByteCount)
+        {
+            Encoding encoding = Encoding.GetEncoding("ascii", new EncoderReplacementFallback("abcde"), DecoderFallback.ExceptionFallback);
+            Assert.Equal(expectedByteCount, encoding.GetByteCount(input));
+        }
+
+        [Fact]
+        public void GetByteCount_WithSingleCharNonAsciiReplacementFallback_ValidatesAscii()
```

Comment:
Note to reviewers: This test case currently fails due to a bug in the current implementation of `ASCIIEncoding`. The actual behavior is that `GetByteCount` returns 1, but correct behavior would instead be for it to throw `ArgumentException`.

I'm working on correcting this issue at the same time that the other `ASCIIEncoding` changes come online.